oak-run/src/main/java/org/apache/jackrabbit/oak/indexversion/PurgeOldIndexVersion.java

```diff
@@ -0,0 +1,226 @@
+        if (!isReadWriteRepository) {
+            log.info("Repository connected in read-only mode. Use '--read-write' for write operations");
+        }
+        try (NodeStoreFixture fixture = NodeStoreFixtureProvider.create(opts)) {
+            NodeStore nodeStore = fixture.getStore();
+            Set<String> indexPathSet = filterIndexPaths(getIndexPathList(nodeStore));
+            Map<String, Set<String>> segregateIndexes = segregateIndexes(indexPathSet);
+            for (Map.Entry<String, Set<String>> entry : segregateIndexes.entrySet()) {
+                String trimmedBaseIndexPath = PurgeOldVersionUtils.trimSlash(entry.getKey());
+                String trimmedIndexParentPath = trimmedBaseIndexPath.substring(0, trimmedBaseIndexPath.lastIndexOf('/'));
+                List<IndexName> indexNameObjectList = getIndexNameObjectList(entry.getValue());
+                NodeState indexDefParentNode = NodeStateUtils.getNode(nodeStore.getRoot(),
+                        "/" + trimmedIndexParentPath);
+                List<IndexVersionOperation> toDeleteIndexNameObjectList =
+                        IndexVersionOperation.applyOperation(indexDefParentNode, indexNameObjectList, this.threshold);
+                if (isReadWriteRepository) {
+                    purgeOldIndexVersion(nodeStore, toDeleteIndexNameObjectList, trimmedBaseIndexPath);
+                } else {
+                    log.info("Repository is opened in read-write mode: IndexOperations" +
+                            " for index at path {} are : {}", trimmedBaseIndexPath, toDeleteIndexNameObjectList);
+                }
+            }
+        }
+    }
+
+    private Map<String, Set<String>> segregateIndexes(Set<String> indexPathSet) {
+        Map<String, Set<String>> segregatedIndexes = new HashMap<>();
+        for (String path : indexPathSet) {
+            String baseIndexPath = getBaseIndexPath(path);
+            Set<String> indexPaths = segregatedIndexes.get(baseIndexPath);
+            if (indexPaths == null) {
+                indexPaths = new HashSet<>();
+            }
+            indexPaths.add(PurgeOldVersionUtils.trimSlash(path));
+            segregatedIndexes.put(baseIndexPath, indexPaths);
+        }
+        return segregatedIndexes;
+    }
+
+    private Iterable<String> getIndexPathList(NodeStore store) throws CommitFailedException, IOException {
+        IndexPathService indexPathService = new IndexPathServiceImpl(store);
+        Iterable<String> indexPaths = indexPathService.getIndexPaths();
+        return indexPaths;
+    }
+
+    private boolean isIndexPathOptionValid(String indexPathOption, String indexPath) {
+        String slashTrimmedIndexPathOption = PurgeOldVersionUtils.trimSlash(indexPathOption);
+        String slashTrimmedIndexPath = PurgeOldVersionUtils.trimSlash(indexPath);
+        if (slashTrimmedIndexPath.startsWith(slashTrimmedIndexPathOption)) {
+            if (slashTrimmedIndexPathOption.length() == slashTrimmedIndexPath.length()
```

Comment:
I would do one comparison at a time, for example:

    if slashTrimmedIndexPathOption.equals(slashTrimmedIndexPath)) {
        return true;
    }
    ...

That way, you get less and less cases as you continue.